src/core/api/coap_api.cpp

```diff
@@ -211,6 +211,8 @@ otError otCoapSendRequestBlockWiseWithParameters(otInstance                 *aIn
     Error                     error;
     const Coap::TxParameters &txParameters = Coap::TxParameters::From(aTxParameters);
 
+    VerifyOrExit(AsCoreType(aMessage).GetOrigin() != Message::kOriginThreadNetif, error = kErrorDrop);
```

Comment:
Is `kErrorInvalidArgs` more appropriate? (Applies to other APIs below)
```suggestion
    VerifyOrExit(AsCoreType(aMessage).GetOrigin() != Message::kOriginThreadNetif, error = kErrorInvalidArgs);
```